### PR TITLE
[CI] Update to ROCm 5.0.2 in dockerfile and remove miopentensile

### PR DESCRIPTION
**Pull request by junliume**

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -759,7 +759,8 @@ if(WORKAROUND_ISSUE_936 AND MIOPEN_TEST_HALF)
     LIST(APPEND IMPLICITGEMM_ARGS --disable-forward --disable-backward-data)
     #Afther fix need to remove '| grep -v "cannot be executed due to incorrect params"'
 endif()
-add_custom_test(test_conv_for_implicit_gemm SKIP_UNLESS_ALL BF16_ENABLED HALF_ENABLED GFX1030_ENABLED
+#TODO Code Quality WORKAROUND ROCm 5.0.2 update
+add_custom_test(test_conv_for_implicit_gemm SKIP_UNLESS_ALL BF16_ENABLED HALF_ENABLED GFX1030_ENABLED GFX90A_DISABLED
```
Comment (atamazov) [inline test/CMakeLists.txt:763]: [Notice] WORKAROUND should refer to the existing github issue or Jira ticket.

**Comment by atamazov** [comment]
@junliume It is ready for review or draft?

**Comment by junliume** [comment]
> @junliume It is ready for review or draft?

It's still a draft :( I am mounting workarounds to make it pass all existing CIs first (and identifying whether it is caused by unstable version of HW). When it is ready I would create the ticket for the WAs.

File: CMakeLists.txt [hunk 2]
```diff
@@ -541,6 +541,17 @@ enable_clang_tidy(
         -readability-qualified-auto
         -readability-redundant-string-init
         -readability-uppercase-literal-suffix
+        #TODO Code Quality WORKAROUND ROCm 5.0.2 update
```
Comment (junliume) [inline CMakeLists.txt:544]: Same here: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1463#discussion_r835345095

**Comment by junliume** [comment]
Update with ROCM 5.1 instead for stability issues

**Pull request closed without merging**